src/pages/wallet/WalletStatementPage.js

```diff
@@ -26,39 +29,96 @@ const propTypes = {
         }).isRequired,
     }).isRequired,
 
+    walletStatement: PropTypes.shape({
+        /** Whether the PDF file available for download or not */
+        isGenerating: PropTypes.bool,
+
+    }),
+
     ...withLocalizePropTypes,
 };
 
-const WalletStatementPage = (props) => {
-    moment.locale(lodashGet(props, 'preferredLocale', 'en'));
-    const yearMonth = lodashGet(props.route.params, 'yearMonth', null);
-    const year = yearMonth.substring(0, 4) || moment().year();
-    const month = yearMonth.substring(4) || moment().month();
-    const monthName = moment(month, 'M').format('MMMM');
-    const title = `${monthName} ${year} statement`;
-    const url = `${CONFIG.EXPENSIFY.EXPENSIFY_URL}statement.php?period=${yearMonth}`;
-    return (
-        <ScreenWrapper>
-            <HeaderWithCloseButton
-                title={Str.recapitalize(title)}
-                shouldShowDownloadButton
-                onCloseButtonPress={() => Navigation.dismissModal(true)}
-                onDownloadButtonPress={() => User.downloadStatementPDF(yearMonth)}
-            />
-            <WalletStatementModal
-                statementPageURL={url}
-            />
-        </ScreenWrapper>
-    );
+const defaultProps = {
+    walletStatement: {
+        isGenerating: false,
+    },
 };
 
+class WalletStatementPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isDownloading: false,
+        };
+        this.processDownload = this.processDownload.bind(this);
+        this.yearMonth = lodashGet(this.props.route.params, 'yearMonth', null);
+    }
+
+    componentDidMount() {
+        User.generateStatementPDF(this.yearMonth);
+    }
```

Comment:
Thinking we should handle some logic here too to make sure we cancel this call to`User.generateStatementPDF` if the user navigates to the component but then quickly navigates away so we don't still download the PDF.

We can make this a cancellable promise and cancel it during `componentWillUnmount`.

Reply:
Good idea, I will add this